sys/include/net/gnrc/sixlowpan/nd/router.h

```diff
@@ -60,6 +60,7 @@ typedef struct gnrc_sixlowpan_nd_router_prf {
     gnrc_ipv6_netif_addr_t *prefix;             /**< prefix on the interface/in the prefix list */
 } gnrc_sixlowpan_nd_router_prf_t;
 
+#ifdef MODULE_GNRC_SIXLOWPAN_ND_ROUTER
```

Comment:
You're right. My thoughts were that this thing doesn't compile without stateful compression enabled (aka gnrc_sixlowpan_ctx) and I thought if it is not needed on non-routers we can just spare it in that case, but yes, that's a) kind of inconsistent and b) actually not a solution. Will fix.